static/app/utils/performance/contexts/pageAlert.tsx

```diff
@@ -0,0 +1,81 @@
+import {createContext, Fragment, useContext, useState} from 'react';
+import {Theme} from '@emotion/react';
+
+import {Alert} from 'sentry/components/alert';
+
+export type PageAlertOptions = {
+  message: React.ReactNode | undefined;
+  type: keyof Theme['alert'];
+};
+
+const pageErrorContext = createContext<{
+  setPageError: (message: string | undefined) => void;
+  setPageInfo: (message: string | undefined) => void;
+  setPageMuted: (message: string | undefined) => void;
+  setPageSuccess: (message: string | undefined) => void;
+  setPageWarning: (message: string | undefined) => void;
+  pageAlert?: PageAlertOptions;
+}>({
+  pageAlert: undefined,
+  setPageError: (_: string | undefined) => {},
+  setPageInfo: (_: string | undefined) => {},
+  setPageMuted: (_: string | undefined) => {},
+  setPageSuccess: (_: string | undefined) => {},
+  setPageWarning: (_: string | undefined) => {},
+});
+
+export function PageAlertProvider({children}: {children: React.ReactNode}) {
+  const [pageAlert, setPageAlert] = useState<PageAlertOptions | undefined>();
+
+  const setPageInfo = (message: string | undefined) => {
+    setPageAlert({message, type: 'info'});
+  };
+  const setPageMuted = (message: string | undefined) => {
+    setPageAlert({message, type: 'muted'});
+  };
+
+  const setPageSuccess = (message: string | undefined) => {
+    setPageAlert({message, type: 'success'});
+  };
+
+  const setPageWarning = (message: string | undefined) => {
+    setPageAlert({message, type: 'warning'});
+  };
+
+  const setPageError = (message: string | undefined) => {
+    setPageAlert({message, type: 'error'});
+  };
```

Comment:
You could clean this repetition up by iterating through the types, and creating functions on the fly!

Reply:
Hmm, I really like the looping idea, but I don't think there's a way to loop through a union type in Typescript.
Perhaps I can just declare a const, and just loop through that?

Reply:
Sorry for the delay! You can loop through union types in TypeScript! It's a little awkward [but possible](https://www.typescriptlang.org/play?target=6#code/MYewdgzgLgBAMgUQGoLgZRgXhgbQOQCWYAZiHjAIYQyiRQA0MeA7hQE5hEDm5VN40RnggBXYMACmECL2q1BTCWzYg2s-nQC6lOQKgBuAFCGoATwAOE+BIBuEgDZYYZyyGLxkqNDjAiAtgBGSppGJhZWcLYOAMIU9vYBFMAA1tTYAN6GMNm4kXb2AHIUflZE1vk6MAAGEBJQAAoUXBIAJOmx5gRQcQQAXhIAPHkORSUAfAC+VZoAXDAAFCXSTRIA-HPQbNwAlFhjMDYgBAAmRhPG8rDAcQlJqU4A8gEAVhLAUAB0xCp+CGBQWyk80QKHQHz8FHM83sUUKxSsmH2mRyGmgNBuiRSowR1VqDRWbWunW69j6EmhsOx2ymlUuRhRl3R8UxySciykEBWGwBOz2C3SMCWnOajBcEjmMPy2Im21CKLYdREHFw12Zd2xjFVtxSIUMMt2fBEYGSYBAzDAlWG9liapSEFChmIRveBHA6OJPX68023K2YC422ROQVUCVFs2H2AAAt2ABBKDzAAM2w+UBAAFVzJY2LFavNdgBqGAR0QBTbceYARlleouAhAMI+9hAXHmWpZEA+eMazQA6uxOP75ngow5m3htrKgA). That's my first stab at it, just to give an idea, maybe a cleaner version is possible